### ISSUE TITLE
Update Android clients

### ISSUE THREAD

**Issue by Monsterovich** [Tox/tox.chat#223]
- Antox is outdated and no longer maintained. 
- Add [Protox](https://gitlab.com/Monsterovich/protox), [Trifa](https://github.com/zoff99/ToxAndroidRefImpl) and Atox (if it's ready) to the list.

**Comment by nurupo** [comment]
Yes, Antox needs to be removed, it seems to be abandoned. Although the repository doesn't state that clearly, the issues users submit are not getting replied/fixed by the developers and there wasn't any commit or a new version released in a while.

As far as adding new clients - it's a bit complicated. [Clients need to pass TCS in order to be added to the client list](https://github.com/Tox/tox.chat#i-want-my-client-to-be-listed-on-your-website). So Protox, Trifa and aTox need to be checked against TCS. The last time someone requested Trifa to be added, [its developer said that it's not compliant with TCS](https://github.com/Tox/tox.chat/issues/163), but that was some time ago, perhaps something has changed since then.

I have reached out to Trifa and aTox devs on IRC, asking if they think their clients are TCS compliant (and ready for general use in case of aTox). Will see what they reply. I was looking for Protox's dev, but it doesn't seem like they hang out on our IRC channels? I noticed [they call their versions "alpha"](https://gitlab.com/Monsterovich/protox/-/blob/master/CHANGELOG), so it's a similar situation to aTox where it might not be ready for general use yet.

**Comment by Monsterovich** [comment]
>Clients need to pass TCS in order to be added to the client list.

TCS has cool things but it should be significantly improved. There are some things with which I don't agree. For example: I don't like TCS avatars because they look bad that's why I implemented [this](https://jdenticon.com/) popular library. My users don't like TCS avatars either. TCS doesn't even have a library which can generate identicons.

>Will see what they reply. I was looking for Protox's dev

Hello. That's me.

**Comment by nurupo** [comment]
>For example: I don't like TCS avatars because they look bad that's why I implemented this popular library.

Hm, looks like the identicon algorithm [was made by @zoff99](https://github.com/zoff99/ToxAndroidRefImpl/blob/zoff99/Identicon_spec_dra/Identicon.md) and then [got added to TCS](https://github.com/Tox/Tox-Client-Standard/pull/35). Looks like no one considered searching for a popular identicon library at the time? Out of curiosity, is there a single-file algorithm for jdention somewhere? All implementations have it split between 20 files.

>There are things that should be added to TCS. For example: mobile Tox clients should be able to export/import profiles or the user should have access to them. That's what Trifa doesn't have.

It does specify an export format, but you are right in that it misses a requirement of being able to import it, which is why you'd want to export in the first place.

By the way, we are reevaluating the requirement of TCS in a neighboring issue https://github.com/Tox/tox.chat/issues/224.

>Hello. That's me.

Oh, hi! Didn't realize it was you, didn't see your username in the GitLab repo, it uses full name everywhere.

I'd say let's keep his issue on hold for now and see how the TCS reevaluation issue plays out.

**Comment by Monsterovich** [comment]
>  is there a single-file algorithm for jdention somewhere

I use [this](https://raw.githubusercontent.com/dmester/jdenticon/master/dist/jdenticon.js) .js file in my QML code.

By the way, can I have access to [this](https://wiki.tox.chat/clients) Wiki page so I can add Protox to it.

**Comment by nurupo** [comment]
Sure, what is your wiki username?

**Comment by Monsterovich** [comment]
> Sure, what is your wiki username?

https://wiki.tox.chat/wiki/user/monsterovich

**Comment by nurupo** [comment]
You should have the needed permissions now.

**Comment by Monsterovich** [comment]
https://tox.chat/clients.html

Can you please mention Protox too?

**Comment by nurupo** [comment]
I have been asking folks over #toktok IRC to look into Protox and share their opinion on it a few times now, back in November and just recently in January when adding aTox to the website, with the intention of making a PR adding Protox to the website, yet both times there were no replies. While the requirement of TCS compliance has been dropped, we can't add a client to the website without anyone having reviewed it and endorsing it.
